core/services/pipeline/common_eth.go

```diff
@@ -0,0 +1,101 @@
+package pipeline
+
+import (
+	"bytes"
+	"regexp"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/pkg/errors"
+)
+
+var (
+	ethABIRegex     = regexp.MustCompile(`\A\s*([a-zA-Z0-9_]+)?\s*\(\s*([a-zA-Z0-9\[\]_\s,]+\s*)?\)`)
+	indexedKeyword  = []byte("indexed")
+	calldataKeyword = []byte("calldata")
+	memoryKeyword   = []byte("memory")
+	storageKeyword  = []byte("storage")
+	spaceDelim      = []byte(" ")
+	commaDelim      = []byte(",")
+)
+
+func parseETHABIArgsString(theABI []byte, isLog bool) (args abi.Arguments, indexedArgs abi.Arguments, _ error) {
+	var argStrs [][]byte
+	if len(bytes.TrimSpace(theABI)) > 0 {
+		argStrs = bytes.Split(theABI, commaDelim)
+	}
+
+	for _, argStr := range argStrs {
+		argStr = bytes.Replace(argStr, calldataKeyword, nil, -1) // Strip `calldata` modifiers
+		argStr = bytes.Replace(argStr, memoryKeyword, nil, -1)   // Strip `memory` modifiers
+		argStr = bytes.Replace(argStr, storageKeyword, nil, -1)  // Strip `storage` modifiers
+		argStr = bytes.TrimSpace(argStr)
+		parts := bytes.Split(argStr, spaceDelim)
+
+		var (
+			argParts [][]byte
+			typeStr  []byte
+			argName  []byte
+			indexed  bool
+		)
+		for i := range parts {
+			parts[i] = bytes.TrimSpace(parts[i])
+			if len(parts[i]) > 0 {
+				argParts = append(argParts, parts[i])
+			}
+		}
+		switch len(argParts) {
```

Comment:
All the handwritten parsing here seems fine but I wonder if we could construct an actual parser for this.

Reply:
Note: can't find a decent parser combinator library for Go so it might be too much of a hassle

Reply:
Yeah the one I'm familiar with seemed like overkill for this

Reply:
We could just put the contract abi json in the spec and then use abigen tools directly to parse it? We already have json elsewhere in our specs and the spec author would already have the contract abi json available

Reply:
We're going to reach out to DevRel to do a survey of devs about this, will move to a subsequent PR if a change is needed